packages/playwright-core/src/client/browserContext.ts

```diff
@@ -100,6 +101,12 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
       if (page)
         page.emit(Events.Page.Console, consoleMessage);
     });
+    this._channel.on('pageError', ({ error, page }) => {
+      this.emit(Events.BrowserContext.PageError, parseError(error));
```

Comment:
So I guess @dgozman wants to see 

```js
this.emit(Events.BrowserContext.PageError, { error: parseError(error), page });
```

here and wants to make sure Page is not nullable.

Reply:
makes sense, @dgozman Do you think is there any path forward to get this in?

Reply:
Is the intention here to expose a `PageError` class, that has client/pageerror and server/pageerror and uses channel dispatcher to be able to be consumed in this way. 

```
context.on("pageerror", pageerror => {
   page.error.page() //  page where error happened
   pageerror.error() // exeception
});
```

Just want to make sure before I make the changes. Thank you in advance.

Reply:
Just `client/pageError.ts`. We don't need server-side `pageError` nor `pageErrorDispatcher`, the way you plumbed through `protocol.yml` looks good. The usage snippet above looks great!

Reply:
Great. Added the necessary changes.